runelite-client/src/main/java/net/runelite/client/plugins/mining/MiningRocksOverlay.java

```diff
@@ -116,13 +136,28 @@ public Dimension render(Graphics2D graphics)
 			ppc.setProgress(percent);
 			ppc.render(graphics);
 		}
-		return null;
+	}
+
+	public void handleCamTorumWaterStreamHighlights(Graphics2D graphics, Map<TileObject, Tile> streams)
+	{
+		streams.forEach((object, tile) ->
+		{
+			Shape objectClickbox = object.getClickbox();
+			if (objectClickbox != null)
+			{
+				Color overlayColor = config.camTorumWaterOverlayColor();
```

Comment:
This fetches config N times per frame, instead of once per frame. Define the color var outside of the loop instead.